src/include/processor/operator/recursive_extend/shortest_path_recursive_join.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "recursive_join.h"
+
+namespace kuzu {
+namespace processor {
+
+class ShortestPathRecursiveJoin : public BaseRecursiveJoin {
+public:
+    ShortestPathRecursiveJoin(uint8_t lowerBound, uint8_t upperBound, storage::NodeTable* nodeTable,
```

Comment:
No we made a decision to purposely use 2 constructors to avoid having NULL child for source of a pipeline. If we change this we should change every single source operator.